blocks.md

```diff
@@ -0,0 +1,80 @@
+
+<h3>Block Plugins and the Block Directory</h3>
+
+The Block Directory contains only Block Plugins, that is to say plugins that contain only a single independent block and a minimum of supporting code. Block plugins have the following characteristics:
+
+1. A specific type of WordPress plugin, with the same structure including a readme.txt file.
+1. Containly only blocks (typically one).
+1. Contain a minimum of server-side code. 
+1. Do not have any UI outside of the post editor.
+
+In addition to following the guidelines that apply to all WordPress plugins, Block Plugins that are submitted to the Block Directory must adhere to these guidelines:
+
+<h3>Guidelines</h3>
+
+<h4>1. Block Plugins are for the Block Editor.</h4>
+
+Your block plugins should contain a block, and a minimum of other supporting code. It should not contain any UX outside of the editor, such as WordPress options or `wp-admin` menus. Server-side code should be kept to a minimum.
+
+<h4>2. Block Plugins are separate blocks.</h4>
+
+Block plugins are intended to be single purpose, separate, independent blocks, not bundles or compendiums of many blocks. In most cases, a Block Plugin should contain only one (single) block. It should not contain multiple blocks that could reasonably be split up into separate, independent plugins.
+
+Block plugins may contain more than one block where a clearly necessary parent/child dependency exists; for example a list block that contains list item blocks.
+
+<h4>3. Plugin titles should reflect the block title.</h4>
+
+We want users to interact with blocks, not product names. Block Plugins should be named to reflect the block; for example "Image Grid" block, "Improved Slider", "Business Hours" block. Please avoid plugin names unrelated to the block itself like "Widgets Incorporated Block", and refrain from block names that include trademarks or names that you do not legally represent; for example you cannot name a block "Facerange Block" unless you work for Facerange and have their permission.
+
+<h4>4. Block Plugins must include a <code>block.json</code> file.</h4>
+
+The Block Registration RFC outlines the format of a `block.json` file: https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md
+
+Block Plugins must include a valid `block.json` file. In addition to the requirements specified in the RFC, the `block.json` must include the following attributes:
+* name
+* title
+* At least one of: script, editorScript
+* At least one of: style, editorStyle
+
+<h4>5. Block Plugins must not require payment or a paid account to function.</h4>
+
+This includes payment for premium features. You may use the donation link feature.
+
+<h4>6. Block Plugins should work independently.</h4>
+
+Block Plugins must not require another plugin in order to function.
+
+They may use an external/cloud API where necessary, but not if that API requires a paid account for access.
```

Comment:
I'm mostly on board with this, but this rule would prevent many useful plugins interfacing with 3P services from being eligible and thus encourage them to go for the traditional plugin route, which would likely result in worse UX.

For example, I think a "MailChimp Subscription Form" block plugin makes complete sense to have, even though you have to pay for the service on the other side. Maybe we could loosen this rule, such as: "If that API or service requires a paid account for access, the service name must be included in the plugin and block name." That would easily make it clear to at least many folks. Of course, at the same time, the trademark rule has to remain satisfied, i.e. if the developer does not work for MailChimp, it could be called "Subscription Form for MailChimp".

Reply:
There are also a number of examples of services that include a free tier, but also have paid tiers, both requiring an account/API token.

Apple & Google's App Stores include a marker that says "Includes In-App Purchases." Usually this is used where there's a baseline, persistent free functionality and a paid upgrade either removes advertisements or adds additional features. However, there are plenty of services that have a free API tier (Mapbox, Google Maps) that max out at a certain usage threshold.

Of course these services can release a standard Plugin to make a Block available, I understand not wanting to be a game-able funnel for paid services to signup new users, but particularly for services that offer a generous but limited free tier I'm not sure this is quite flexible enough.

Reply:
Just a note on Felix's idea, we can't require plugin names to have trademarks (the name of the external service) in them, as the Plugins team already is dealing with companies who want plugins removed that have trademarks of theirs anywhere in the title (example: facebook)

Reply:
Take my Charterboat plugin for example. One of the "big wins" I've been working on with industry leaders is innovate a seamless integration for independent Charterboats that rely on bookings through outside travel agents. I've created a series of blocks which will read from the boat's REST API and produce an availability calendar, boat listing, etc by only inputing the domain of the remote Charterboat. 

This would really benefit the industry as a whole, but I read this rule to say that these blocks could not be admitted to the directory because they require that the boat has installed the premium plugin Charter Boat Bookings Pro in order to have that REST Data available to the block?

Reply:
> This would really benefit the industry as a whole, but I read this rule to say that these blocks could not be admitted to the directory because they require that the boat has installed the premium plugin Charter Boat Bookings Pro in order to have that REST Data available to the block?

Hey Meg, you can still add the plugin to the directory - just not the 'block directory'. Plugins for the block directory are different to other plugins like yours. A 'block directory' plugin contains only a single block - compltely self-contained with no external dependancies beyond what is in WordPress. This is so they can be installed on-the-fly as single blocks without need to install anything more.

Your plugin contains several blocks so it is more suited for the standard plugin directory :)

Reply:
That will be a repetitive point to make "you can still add the plugin to the directory - just not the 'block directory'." and should be as often in the documentation as possible.